it/controllers/benefits/reimbursed/NonTaxableCostsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,540 @@
+        }
+
+        //          TODO: This will need updating to the next page in the flow when the navigation is hooked up
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe
+            Some(s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
+        }
+
+        "updates non taxable costs question to no and non taxable costs to none" in {
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expensesQuestion)) shouldBe Some(false)
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expenses)) shouldBe None
+        }
+
+      }
+
+      "Update the non taxable costs question to yes, redirects to the non taxable costs amount page" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullReimbursedCostsVouchersAndNonCashModel.copy(expensesQuestion = Some(false)))))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the non taxable costs amount page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe
+            //          TODO: This will be updated to the non taxable costs amount page when its created
+            Some(s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
+        }
+
+        "updates non taxable costs question to yes" in {
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expensesQuestion)) shouldBe Some(true)
+        }
+
+      }
+
+      "redirect the user to the overview page when it is in year" which {
+        lazy val result: WSResponse = {
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(pageUrl(taxYear), body = "", user.isWelsh, follow = false,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/update-and-submit-income-tax-return/$taxYear/view")
```

Comment:
This can be changed 
shouldBe Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))